staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go

```diff
@@ -897,16 +897,12 @@ func (ss *scaleSet) EnsureHostInPool(service *v1.Service, nodeName types.NodeNam
 	defer cancel()
 	klog.V(2).Infof("EnsureHostInPool begins to update vmssVM(%s) with new backendPoolID %s", vmName, backendPoolID)
 	rerr := ss.VirtualMachineScaleSetVMsClient.Update(ctx, nodeResourceGroup, ssName, instanceID, newVM, "network_update")
-	if rerr != nil && rerr.Retriable && ss.CloudProviderBackoff {
```

Comment:
where is the retry logic? I don't see armclient is used some where in this PR?

Reply:
See https://github.com/kubernetes/kubernetes/pull/86740/files#diff-9c5eab3304e3867d8a50d9b7ff7bb8e6R101-R112 for the retries. `retry.DoExponentialBackoffRetry()` is used as a decorator for backoff retries.